File: Go/euler-135-same-differences.go/euler-135-same-differences.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

// Function which finds finds the number of solutions for x^2 - y^2 - z^2 by looking at the difference variable (d) and itterating over each combination.
// At a computational level we are essentially marking each solution when we reach it, then returning all the solutions we reached 10 times.
func findSolutions(n int) int {
	solutions := make([]int, n)
	for x := 1; x < n*2; x++ {
		for d := x/5 + 1; d < (x+1)/2; d++ {
			tmp := (x - d) * (d*5 - x)
			if tmp >= n {
				break
			}
			solutions[tmp]++
		}
	}
	answer := 0
	// Count the number of solutions x for each n where x == 10
	for _, count := range solutions {
		if count == 10 {
			answer++
		}
	}
	return answer
}

func main() {
	args := os.Args

	for i := 1; i < len(args); i++{
		n, _ := strconv.Atoi(args[i])
		fmt.Println("There are", findSolutions(n), "values less than", n, "that have exactly ten distinct solutions")
}
}

File: Java/euler_135_same_differences.java/euler_135_same_differences.java
public class euler_135_same_differences {

    // function which takes a given n limit and finds the solutions which occur 10 times and returns the count of how many it found..
    private static int findSolutions(int n) {
        int[] solutions = new int[n];
        for (int x = 1; x < n * 2; x++) {
            for (int d = x / 5 + 1; d < (x + 1) / 2; d++) {
                int tmp = (x - d) * (d * 5 - x);
                if (tmp >= n) {
                    break;
                }
                solutions[tmp]++;
            }
        }
        int answer = 0;
        for (int count : solutions) {
            if (count == 10) {
                answer++;
            }
        }
        return answer;
    }
    
    // Main function takes user input and calls the function against it.
    public static void main(String[] args) {
        for (int i = 0; i < args.length; i++) {
            int n = Integer.parseInt(args[i]);
            System.out.println("There are " + findSolutions(n) + " values less than " + n + " that have exactly ten distinct solutions");
        }
    }
}
